Automattic/wp-calypso
# Update/move purchase overlap to thank you page

## Proposed Changes

* This diff moves messaging for users with multiple Akismet purchases out of the cart and onto the thank you page

## Testing Instructions

* You will need an account that has an Akismet Pro 10K subscription. If you have other higher-tier subscriptions on the account, cancel or remove them before proceeding
* Before applying the patch (or by accessing the production checkout), you can observe the current behavior. If you have the 10K plan and add the 20K plan to the cart, you will get the message shown below. URL to add 20K to the cart: https://wordpress.com/checkout/akismet/ak_plus_yearly_2
![Screenshot 2023-05-02 at 5 06 32 PM](https://user-images.githubusercontent.com/18016357/235787513-9095959c-779f-4da4-8044-196f161b2a87.png)
* After completing any Akismet purchase, currently, the thank you screen shows this messaging:
![Screenshot 2023-05-02 at 5 09 32 PM](https://user-images.githubusercontent.com/18016357/235787618-939f3692-58de-4f8c-836d-1a76de46391f.png)
* With the patch applied, notice that adding a higher-tier product to your cart no longer shows the product overlap notice in the cart: http://calypso.localhost:3000/checkout/akismet/ak_plus_yearly_2
![Screenshot 2023-05-02 at 5 06 39 PM](https://user-images.githubusercontent.com/18016357/235787787-a35534f9-3204-4cc1-836e-dec04c11fd99.png)
* On the thank you screen, if the purchase has an API limit and the user has more than one Akismet plan, the messaging now lets the user know that they should check their account page to remove any subscriptions that they do not need. You can view this screen directly at this URL: http://calypso.localhost:3000/checkout/akismet/thank-you/ak_plus_yearly_2
![Screenshot 2023-05-02 at 5 09 25 PM](https://user-images.githubusercontent.com/18016357/235788056-ec84021e-efe9-4b1a-aa35-de381a6b2463.png)
* If the purchase does not have a stated API limit (free or personal plans) then the thank you screen shows the "upgrade" messaging without the text that lets the user know to cancel other plans
![Screenshot 2023-05-02 at 5 09 54 PM](https://user-images.githubusercontent.com/18016357/235788405-e7f23d98-487b-4046-a585-0c0f6e73de18.png)
*If the purchase is the first Akismet purchase on the account, then the current "Let the spam blocking party begin" messaging is shown on the thank you screen.

Note, there is a small edge case where a user who has a Pro or Business plan and adds a personal plan will not see the multiple-plans messaging on the thank you screen. This seems like an unlikely "upgrade" path, so it has not been included in the logic.

## Pre-merge Checklist
